cpp/CMakeLists.txt

```diff
@@ -457,7 +456,10 @@ if(RAFT_COMPILE_LIBRARY)
                INTERFACE_POSITION_INDEPENDENT_CODE ON
   )
 
-  target_link_libraries(raft_lib PUBLIC raft::raft $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>)
+  target_link_libraries(
+    raft_lib PUBLIC raft::raft ${RAFT_CUBLAS_DEPENDENCY}
```

Comment:
Why does the raft lib require cublas but not the others? Why are the others only specified as interface dependencies below?

Reply:
Ideally we would only be linking against dependencies which are actually needed for each of the targets- cublas being the only dependency which is explicitly needed for the APIs compiled into `libraft`. The problem is that until we implement #1394, we have to depend on all the math libraries because `device_resources` (implicitly) requires all of the handles be available.